### PR TITLE
[+] Databases Connections - Support multiple databases connections

### PR DESCRIPTION
**Pull request by SeyZ**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by SeyZ** [comment]
models/index.js
```
'use strict';
var fs = require('fs');
var path = require('path');
var Sequelize = require('sequelize');

let dbs = [{
  name: 'forest_movies',
  connectionString: process.env.MOVIES_DATABASE_URL
}, {
  name: 'forest_meals',
  connectionString: process.env.MEALS_DATABASE_URL
}];

let sequelize = {};
let db = {};

dbs.forEach((info) => {
  let databaseOptions = {
    logging: console.log,
    pool: { maxConnections: 10, minConnections: 1 },
    dialectOptions: {}
  };

  if (process.env.SSL_DATABASE) {
    databaseOptions.dialectOptions.ssl = true;
  }

  if (process.env.ENCRYPT_DATABASE) {
    databaseOptions.dialectOptions.encrypt = true;
  }


  let connection = new Sequelize(info.connectionString, databaseOptions);
  sequelize[info.name] = connection;

  fs
    .readdirSync(path.join(__dirname,info.name))
    .filter(function (file) {
      return (file.indexOf('.') !== 0) && (file !== 'index.js');
    })
    .forEach(function (file) {
      try {
        var model = connection['import'](path.join(__dirname, info.name, file));
        sequelize[model.name] = model;
      } catch (error) {
        console.error('Model creation error: ' + error);
      }
    });

  Object.keys(connection).forEach(function(modelName) {
    if ('associate' in connection[modelName]) {
      sequelize[modelName].associate(sequelize);
    }
  });
});

db.Sequelize = Sequelize;
db.sequelize = sequelize;

module.exports = db;
```

app.js
```
// ...
app.use(require('forest-express-sequelize').init({
  envSecret: process.env.FOREST_ENV_SECRET,
  authSecret: process.env.FOREST_AUTH_SECRET,
  sequelize: [
    models.sequelize['forest_movies'],
    models.sequelize['forest_meals']
  ]
}));

//...
```

Ensuite, le repertoire models doit avoir cette structure:
`models/forest_movies/movie.js`
`models/forest_meals/product.js`
etc.

**Comment by SeyZ** [comment]
[Oyst want to have a Lumber generated admin with multi databases support](https://trello.com/c/0bb74tZm/1727-oyst-want-to-have-a-lumber-generated-admin-with-multi-databases-support)

File: index.js [hunk 1]
```diff
@@ -9,8 +10,12 @@ exports.StatSerializer = Interface.StatSerializer;
 exports.ResourceSerializer = Interface.ResourceSerializer;
 exports.ResourceDeserializer = Interface.ResourceDeserializer;
 exports.Schemas = Interface.Schemas;
+exports.ResourcesRoute = Interface.ResourcesRoute;
 
 exports.init = function(opts) {
+  opts.sequelizeFct = _.isArray(opts.sequelize) ? opts.sequelize[0] :
```
[the remark this answers is not in the document]
Comment (arnaudbesnier) [reply index.js]: And by the way I find quite confusing that `modelsDir` is not mandatory and that models can be retrieved in `sequelize`.

File: services/search-builder.js [hunk 2]
```diff
@@ -102,22 +102,22 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
...
                     parseInt(params.search, 10) || 0);
                 }
               } else if (params.search.match(REGEX_UUID)) {
-                q = opts.sequelize.where(column, ' = ', params.search);
+                q = opts.sequelizeFct.where(column, ' = ', params.search);
               }
             } else if (field.type === 'String') {
-              q = opts.sequelize.where(
-                opts.sequelize.fn('lower', column), ' LIKE ',
-                opts.sequelize.fn('lower', '%' + params.search + '%')
```
Comment (arnaudbesnier) [inline services/search-builder.js]: All theses identical changes, show that the current implementation is not that great.
In a perfect world, we should not have to replace all these `opts.sequelize` by `opts.sequelizeFct`.

File: index.js [hunk 3]
```diff
@@ -9,8 +10,12 @@ exports.StatSerializer = Interface.StatSerializer;
 exports.ResourceSerializer = Interface.ResourceSerializer;
 exports.ResourceDeserializer = Interface.ResourceDeserializer;
 exports.Schemas = Interface.Schemas;
+exports.ResourcesRoute = Interface.ResourcesRoute;
```
Comment (arnaudbesnier) [inline index.js]: What is it for? I saw the PR for that: https://github.com/ForestAdmin/forest-express/pull/97/files

File: index.js [hunk 4]
```diff
@@ -25,20 +30,36 @@ exports.init = function(opts) {
...
   };
 
   exports.SchemaAdapter = require('./adapters/sequelize');
 
   exports.getModels = function () {
-    return opts.sequelize ? opts.sequelize.models : [];
+    var models = [];
+
+    if (_.isArray(opts.sequelize)) {
+      _.each(opts.sequelize, function (db) {
```
[the remark this answers is not in the document]
Comment (SeyZ) [reply index.js]: yes

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (SeyZ) [reply index.js]: • Yes, sequelizeFct is all the functions sequelize provides. There are helpers like "col", "where", etc. 
• If we change the snippet with the ORM param and the list of connections, we just need to ensure the backward compatibility. Let me think about that

File: index.js (hunk 3 above)
Comment (SeyZ) [reply index.js]: From my app, I only have `forest-express-sequelize` and not `forest-express` so this is a a kind "proxy exports" :D Make sense?

Comment (arnaudbesnier) [reply index.js]: ok but why do you need this for you app?

Comment (SeyZ) [reply index.js]: I've discovered an easiest way to implement hasmany relationships:

```
var express = require('express');
var router = express.Router();
var liana = require('forest-express-sequelize');

function orders(req, res, next) {
  req.query.filterType = 'and';
  req.query.filter = { merchant_id: req.params.merchantId };

  liana.ResourcesRoute.order.list(req, res, next);
}

router.get('/merchant/:merchantId/relationships/orders', orders);

module.exports = router;
```

Comment (arnaudbesnier) [reply index.js]: ok, why not, I didn't tested it.
But it is not related to the multiple connections database, right?

File: index.js [hunk 5]
```diff
@@ -25,20 +33,32 @@ exports.init = function(opts) {
...
 
   exports.SchemaAdapter = require('./adapters/sequelize');
 
   exports.getModels = function () {
-    return opts.sequelize ? opts.sequelize.models : [];
+    var models = {};
+
+    _.each(opts.connections, function (connection) {
+      _.each(connection.models, function (model) {
+        models[model.name] = model;
```
Comment (arnaudbesnier) [inline index.js]: I changed that to fix the integrations support (hash vs array)

**Pull request merged**